lib/core/versions/0.5.0/OperationProcessor.ts

```diff
@@ -20,81 +20,77 @@ export default class OperationProcessor implements IOperationProcessor {
     didDocumentReference: { didDocument: object | undefined }
   ): Promise<ApplyResult> {
-    let operationHash = undefined;
-
     try {
       // NOTE: only used for read interally.
       const didDocument = didDocumentReference.didDocument as (DocumentModel | undefined);
-
       const operation = AnchoredOperation.createAnchoredOperation(anchoredOperationModel);
-      operationHash = operation.operationHash;
 
       if (operation.type === OperationType.Create) {
 
         // If we have seen a previous create operation.
         if (didDocumentReference.didDocument) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         const originalDidDocument = Document.from(operation.encodedPayload, this.didMethodName, ProtocolParameters.hashAlgorithmInMultihashCode)!;
 
         const signingKey = Document.getPublicKey(originalDidDocument, operation.signingKeyId);
 
         if (!signingKey) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         if (!(await operation.verifySignature(signingKey))) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         didDocumentReference.didDocument = originalDidDocument;
-        return { validOperation: true, operationHash };
+        return { validOperation: true };
       } else if (operation.type === OperationType.Delete) {
         // Delete can be applied only on valid did with a current document
         if (!didDocument) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         // The current did document should contain the public key mentioned in the operation ...
         const publicKey = Document.getPublicKey(didDocument, operation.signingKeyId);
         if (!publicKey) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         // ... and the signature should verify
         if (!(await operation.verifySignature(publicKey))) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         // If the delete is valid
         didDocumentReference.didDocument = undefined;
-        return { validOperation: true, operationHash };
+        return { validOperation: true };
       } else {
         // Update operation
 
         // If we have not seen a valid create operation yet.
         if (didDocument === undefined) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         // The current did document should contain the public key mentioned in the operation ...
         const publicKey = Document.getPublicKey(didDocument, operation.signingKeyId);
         if (!publicKey) {
```

Comment:
nit: `if (!publickey || !(await operation.verifySignature(publicKey)))` ? Same with other places that used the same structure.

Reply:
Same comment as other versioned code.